app/views/clients/viewclient.html

```diff
@@ -37,16 +37,31 @@ <h3 class="bolder">{{'label.button.deletesig' | translate}}</h3>
 		<div class="modal-header silver">
 			<img ng-src="{{largeImage}}" alt="Signature">
 		</div>
-		<div class="modal-body ">
+		<div class="modal-body">
+			<api-validate></api-validate>
+			{{'label.heading.selectclientsignature' | translate}}
+			<br>
+			<input type="file" ngf-select ng-model="file">
+		</div>
+		<div class="modal-footer">
 			<button class="btn btn-warning" ng-click="cancel()">{{'label.button.cancel' | translate}}</button>
+			<button class="btn btn-danger" ng-click="deleteSig()">{{'label.button.delete' | translate}}</button>
```

Comment:
`has-permission` attribute is required, we can't allow any user to just upload and delete signatures. Also, add a nice trash icon before the `Delete` Text